include/circt/Dialect/HW/InnerSymbolTable.h

```diff
@@ -33,12 +37,16 @@ class InnerSymTarget {
   explicit InnerSymTarget(Operation *op) : InnerSymTarget(op, 0) {}
 
   /// Target an operation and a field (=0 means the op itself).
-  InnerSymTarget(Operation *op, size_t fieldID)
-      : op(op), portIdx(invalidPort), fieldID(fieldID) {}
+  InnerSymTarget(Operation *op, size_t fieldID,
+                 Operation *symbolTableOp = nullptr)
```

Comment:
This seems unsafe.  The op with a symboltable on it may not be legal to access by users of an InnerSymTarget in the context in which they are running.

Reply:
By context, do you mean in a parallelization context?

I think this is the doublely-linked stuff @mortbopet was referring to in the PR comments, yes? If so, I think it makes sense to just ditch it.

Reply:
Ditched!